File: backend/app/models/email.py
from enum import Enum
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class EmailKategorija(str, Enum):
    RFQ = "RFQ"
    NAROCILO = "Naročilo"
    SPREMEMBA = "Sprememba"
    DOKUMENTACIJA = "Dokumentacija"
    REKLAMACIJA = "Reklamacija"
    SPLOSNO = "Splošno"


class RfqPodkategorija(str, Enum):
    KOMPLETNO = "Kompletno"
    NEPOPOLNO = "Nepopolno"
    POVPRASEVANJE = "Povpraševanje"
    REPEAT_ORDER = "Repeat Order"


class EmailStatus(str, Enum):
    NOV = "Nov"
    PREBRAN = "Prebran"
    DODELJEN = "Dodeljen"
    OBDELAN = "Obdelan"


class EmailBase(BaseModel):
    zadeva: str
    posiljatelj: str
    prejemniki: str
    telo: Optional[str] = None


class EmailCreate(EmailBase):
    outlook_id: str
    kategorija: EmailKategorija = EmailKategorija.SPLOSNO
    izvleceni_podatki: Optional[dict] = None


class EmailUpdate(BaseModel):
    projekt_id: Optional[int] = None
    kategorija: Optional[EmailKategorija] = None
    status: Optional[EmailStatus] = None
    rfq_podkategorija: Optional[RfqPodkategorija] = None


class Email(EmailBase):
    id: int
    outlook_id: str
    projekt_id: Optional[int] = None
    kategorija: EmailKategorija
    rfq_podkategorija: Optional[RfqPodkategorija] = None
    status: EmailStatus = EmailStatus.NOV
    datum: datetime
    izvleceni_podatki: Optional[dict] = None
    priloge: Optional[list[str]] = None
    analiza_status: Optional[str] = None
    analiza_rezultat: Optional[dict] = None

    class Config:
        from_attributes = True


class EmailAnalysis(BaseModel):
    """Rezultat LLM analize emaila"""
    kategorija: EmailKategorija
    rfq_podkategorija: Optional[RfqPodkategorija] = None
    zaupanje: float  # 0-1
    izvleceni_podatki: dict
    predlagan_projekt_id: Optional[int] = None
    povzetek: str

File: flutter_app/lib/models/email.dart
/// Model emaila
class Email {
  final int id;
  final String? zadeva;
  final String? posiljatelj;
  final String? prejemniki;
  final String? kategorija;
  final String? rfqPodkategorija;
  final String? status;
  final DateTime? datum;
  final String? analizaStatus;
  final Map<String, dynamic>? analizaRezultat;
  final List<dynamic>? priloge;
  final Map<String, dynamic>? izvleceniPodatki;

  Email({
    required this.id,
    this.zadeva,
    this.posiljatelj,
    this.prejemniki,
    this.kategorija,
    this.rfqPodkategorija,
    this.status,
    this.datum,
    this.analizaStatus,
    this.analizaRezultat,
    this.priloge,
    this.izvleceniPodatki,
  });

  factory Email.fromJson(Map<String, dynamic> json) {
    return Email(
      id: json['id'],
      zadeva: json['zadeva'],
      posiljatelj: json['posiljatelj'],
      prejemniki: json['prejemniki'],
      kategorija: json['kategorija'],
      rfqPodkategorija: json['rfq_podkategorija'],
      status: json['status'],
      datum: json['datum'] != null ? DateTime.parse(json['datum']) : null,
      analizaStatus: json['analiza_status'],
      analizaRezultat: json['analiza_rezultat'] != null
          ? Map<String, dynamic>.from(json['analiza_rezultat'])
          : null,
      priloge: json['priloge'] != null
          ? List<dynamic>.from(json['priloge'])
          : null,
      izvleceniPodatki: json['izvleceni_podatki'] != null
          ? Map<String, dynamic>.from(json['izvleceni_podatki'])
          : null,
    );
  }

  /// Ali ima priloge
  bool get imaPriloge => priloge != null && priloge!.isNotEmpty;

  /// Število prilog
  int get steviloPrilog => priloge?.length ?? 0;

  /// Prikaz kategorije z pod-kategorijo za RFQ
  String get kategorijaDisplay {
    if (kategorija == 'RFQ' && rfqPodkategorija != null) {
      return 'RFQ - $rfqPodkategorija';
    }
    return kategorija ?? '';
  }
}
